src/tools/idlc/src/plugin.c

```diff
@@ -105,17 +127,34 @@ idlc_load_generator(idlc_generator_plugin_t *plugin, const char *lang)
     }
   }
 
-  if ((handle = openlib(path)) || (lang != path && (handle = openlib(lang)))) {
+  /* open the library */
+  if (lang == path) {
+    handle = openlib(path);
+  } else {
+    handle = openlib(lang);
+    if (handle) {
+      path = lang;
+    } else {
+      handle = openlib(path);
+    }
+  }
+  if (handle) {
```

Comment:
I think trying to `openlib(path)` is more than enough since `path` is assigned the "generated" library name.

Reply:
Yes I agree. I'll change that.